app/models/manageiq/providers/redhat/infra_manager/vm_or_template_shared/scanning.rb

```diff
@@ -28,8 +28,81 @@ def perform_metadata_sync(ost)
     sync_stashed_metadata(ost)
   end
 
+  def proxies4job(job = nil)
+    _log.debug "Enter (RHEVM)"
+    msg = 'Perform SmartState Analysis on this VM'
+    embedded_msg = nil
+
+    # If we do not get passed an model object assume it is a job guid
+    if job && !job.kind_of?(ActiveRecord::Base)
+      jobid = job
+      job = Job.find_by_guid(jobid)
+    end
+
+    all_proxy_list = storage2proxies
+    proxies = storage2active_proxies(all_proxy_list)
+    _log.debug "# proxies = #{proxies.length}"
+
+    if proxies.empty?
+      msg = embedded_msg.nil? ? 'No active SmartProxies found to analyze this VM' : embedded_msg
```

Comment:
@roliveri `embedded_msg` is never used and should be removed which would simplify this logic.
`msg = 'No active SmartProxies found to analyze this VM' if proxies.empty?`